OpenRA.Game/Network/OrderIO.cs

```diff
@@ -54,28 +54,25 @@ IEnumerable<Order> ParseData(World world)
 
 		public byte[] Serialize(int frame)
 		{
-			if (data != null)
-				return data.ToArray();
-
-			var ms = new MemoryStream();
+			var ms = new MemoryStream((int)data.Length + 4);
 			ms.WriteArray(BitConverter.GetBytes(frame));
-			foreach (var o in orders)
-				ms.WriteArray(o.Serialize());
-			return ms.ToArray();
+
+			data.Position = 0;
+			data.CopyTo(ms);
+
+			return ms.GetBuffer();
 		}
 
 		public static OrderPacket Combine(IEnumerable<OrderPacket> packets)
 		{
-			var orders = new List<Order>();
+			var ms = new MemoryStream();
```

Comment:
```suggestion
			using (var ms = new MemoryStream())
```

I'm wondering, don't we need `using` around these or does it already dispose immediately after the function has ran